examples/validate-fee/modules/shared_data/src/main/scala/com/my/validate_fee/shared_data/errors/Errors.scala

```diff
@@ -0,0 +1,25 @@
+package com.my.validate_fee.shared_data.errors
+
+import cats.syntax.all._
+import org.tessellation.currency.dataApplication.DataApplicationValidationError
+import org.tessellation.currency.dataApplication.dataApplication.DataApplicationValidationErrorOr
+import org.tessellation.schema.balance.Amount
+
+object Errors {
+  type DataApplicationValidationType = DataApplicationValidationErrorOr[Unit]
+
+  val valid: DataApplicationValidationType = ().validNec[DataApplicationValidationError]
+
+  implicit class DataApplicationValidationTypeOps[E <: DataApplicationValidationError](err: E) {
+    def invalid: DataApplicationValidationType = err.invalidNec[Unit]
+
+    def unlessA(cond: Boolean): DataApplicationValidationType = if (cond) valid else invalid
+
+    def whenA(cond: Boolean): DataApplicationValidationType = if (cond) invalid else valid
+  }
+
+  case class UnexpectedAmount(name: String, expected: Amount, actual: Amount) extends DataApplicationValidationError {
```

Comment:
I think this should use the `DataApplicationFeeError` in place of `DataApplicationValidationError`

Reply:
Done; will appear in the next push.